src/commands/php-errors.php

```diff
@@ -73,16 +73,43 @@ protected function execute( InputInterface $input, OutputInterface $output ) {
         $output->writeln( "<comment>Adding temporary bot collaborator to $site_domain.</comment>" );
 
         $add_collaborator_request = $api_helper->call_pressable_api(
-                "sites/{$pressable_site->id}/collaborators",
+                "collaborators/batch_create",
                 'POST',
                 array(
-                    'siteId' => $pressable_site->id,
-                    'email' => PRESSABLE_BOT_COLLABORATOR_EMAIL,
+                    'siteIds' => array(
+                        $pressable_site->id,
+                    ),
+                    'email'   => PRESSABLE_BOT_COLLABORATOR_EMAIL,
+                    'roles'   => 'sftp_access',
                 )
         );
 
-        if ( empty( $add_collaborator_request->data->accountId ) ) {
-            $output->writeln( "<error>Failed to create a temporary bot collaborator. Aborting!</error>" );
+        if ( ! is_null( $add_collaborator_request->errors ) ) {
+            $output->writeln( "<error>Error creating temporary bot collaborator. Aborting!</error>" );
+            exit;
+        }
+        // Get new collaborator id.
+        $max_tries = 0;
+        while ( empty( $collaborator_id ) || 3 >= $max_tries ) {
+
+            $get_collaborator_list = $api_helper->call_pressable_api(
+                    "/sites/{$pressable_site->id}/collaborators",
+                    'GET',
+                    array()
+            );
+
+            foreach ( $get_collaborator_list->data as $collaborator ) {
+                if ( PRESSABLE_BOT_COLLABORATOR_EMAIL === $collaborator->email ) {
+                    $collaborator_id = $collaborator->id;
+                    break( 2 );
+                }
+            }
+            sleep( 1 );
+            $max_tries++;
+        }
+
+        if ( empty( $collaborator_id ) ) {
```

Comment:
If it can't find the  `$collaborator_id` in 3 tries, execution will stop. What if the `$collaborator_id` takes a few more seconds to become available and then we end up with a ghost user on the site that we will never remove?

It's probably ok, but good to keep in mind.

Reply:
> In other words, if we're unable to delete the temp user, the user will be removed the next time the command is used

Good to know! :)